book/src/pong-tutorial/pong-tutorial-03.md

```diff
@@ -116,190 +116,328 @@ We're finally ready to implement the `PaddleSystem` in `systems/paddle.rs`:
+entities and their component data. `input` is the `InputHandler` resource, which we asked for
+above with `.read_resource::<InputHandler>()`. Finally, `query_paddles` contains the query
+that we described above with `.with_query`. Our closure has all of the data we requested,
+and it's time to write the system's behavior!
+
+```rust
+# mod pong {
+#   pub enum Side {
+#       Left,
+#       Right,
+#   }
+#   pub struct Paddle {
+#       pub side: Side,
+#   }
+# 
+#   pub const ARENA_HEIGHT: f32 = 100.0;
+#   pub const PADDLE_HEIGHT: f32 = 16.0;
+# }
+# 
+# use amethyst::{
+#     core::transform::Transform,
+#     ecs::SystemBuilder,
+#     input::{get_input_axis_simple, InputHandler},
+#     prelude::*,
+# };
+# 
+# // You'll have to mark PADDLE_HEIGHT as public in pong.rs
+# use crate::pong::{Paddle, Side, ARENA_HEIGHT, PADDLE_HEIGHT};
+# 
+# pub struct PaddleSystem;
+# 
+# impl System for PaddleSystem {
+#    fn build(self) -> Box<dyn ParallelRunnable> {
+#        Box::new(
+            SystemBuilder::new("PaddleSystem")
+                .with_query(<(&Paddle, &mut Transform)>::query())
+                .read_resource::<InputHandler>()
+                .build(move |_commands, world, input, query_paddles| {
+                    for (paddle, transform) in query_paddles.iter_mut(world) {
+                        // read input
+                        // move paddle
+                    }
+                }),
+#        )
+#    }
+# }
+# fn main() {}
+```
 
 Now that we have access to the storages of the components we want, we can iterate
-over them. We perform a join operation over the `Transform` and `Paddle`
-storages. This will iterate over all entities that have both a `Paddle`
-and `Transform` attached to them, and give us access to the actual components,
-immutable for the `Paddle` and mutable for the `Transform`.
+over them. We invoke the `iter_mut` method on our query to perform a join operation 
+over the `Transform` and `Paddle` storages. This will iterate over all entities that 
+have both a `Paddle` and `Transform` attached to them, and give us access to the actual components, immutable for the `Paddle` and mutable for the `Transform`.
 
 > There are many other ways to use storages. For example, you can use them to get
 > a reference to the component of a specific type held by an entity, or simply
 > iterate over them without joining. However, in practice, your most common use will
 > be to join over multiple storages as it is rare to have a system affect
 > only one specific component.
 
-> Please also note that it is possible to join over storages using multiple threads
-> by using `par_join` instead of `join`, but here the overhead introduced is not
-> worth the gain offered by parallelism.
+## Reading the input
 
-Let's add this system to our `DispatcherBuilder` in `main.rs`:
```

Comment:
Moved the `main.rs` change farther down in the tutorial. Added a separate section for using `InputHandler`.